docs/reference/contributing/connectivity/WifiInterface.md

```diff
@@ -1,2 +1,44 @@
-## WiFiInterface
+<h2 id="wifi-port">Wi-Fi</h2>
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/wifi.png)<span>Wi-Fi driver</span></span>
+
+This document describes how to port and test an IEEE 802.11 Wi-Fi driver to Mbed OS. There are two types of Wi-Fi drivers in Mbed OS, depending on which protocol layer it implements. Wi-Fi drivers are either a special case of Ethernet driver or external IP stacks. Wi-Fi drivers require configuration from an application and, therefore, implement both the low-level EMAC API or network stack API and the high-level controlling interface API called `WiFiInterface`.
+
+### WifiInterface
+
+The `WifiInterface` class is the controlling API from application to driver.
+You can use this API for configuring security settings, network names and keys and also instantiating connection or disconnection phases from the network.
+
+Whether the driver is an AT-command driven external IP stack or an Ethernet device, this is the API to implement.
+
+The driver takes care of the network joining process and presents Mbed OS only Ethernet layer interface, or IP stack. The driver and the device itself do all of the security handshakes.
+
```

Comment:
"..and presents Mbed OS only Ethernet layer interface, or IP stack" - this line is not very clear to me. What does it mean..that it presents the WiFi device as an ETH layer interface or an IP stack (3rd party driver).. or something else?

Reply:
Added link to Technology/IP networking section. This separation of different layer WiFi drivers is described there.